src/WebJobs.Script.WebHost/Diagnostics/DiagnosticEventTableStorageRepository.cs

```diff
@@ -41,34 +42,36 @@ public class DiagnosticEventTableStorageRepository : IDiagnosticEventRepository,
         private string _tableName;
 
         internal DiagnosticEventTableStorageRepository(IConfiguration configuration, IHostIdProvider hostIdProvider, IEnvironment environment, IScriptHostManager scriptHostManager,
-            ILogger<DiagnosticEventTableStorageRepository> logger, int logFlushInterval)
+            ILogger<DiagnosticEventTableStorageRepository> logger, AzureComponentFactory componentFactory, AzureEventSourceLogForwarder logForwarder, int logFlushInterval)
```

Comment:
I'd prefer here to avoid passing `AzureComponentFactory `and `AzureEventSourceLogForwarder` to create the `TableServiceClientProvider`. Instead we could study (doesn't necessarily need to be in this PR) the creation of a `HostAzureTableStorageProvider` like we do with Blobs: https://github.com/Azure/azure-functions-host/blob/59d8da491072178261050006894f0db0f48951e2/src/WebJobs.Script/StorageProvider/HostAzureBlobStorageProvider.cs 

Is there any implication on not unifying configuration as we do in `HostAzureBlobStorageProvider`?

Reply:
I like this idea. There’s no need to pass AzureEventSourceLogForwarder around since we only need to start it once. We can implement it this way without passing it around.
```
services.AddSingleton<AzureEventSourceLogForwarder>(provider => AzureEventSourceLogForwarder.Start());
```